pysilcam/__main__.py

```diff
@@ -657,47 +675,46 @@ def configure_logger(settings):
 
 
-def adminSTATS(logger, settings, overwriteSTATS, datafilename, datapath):
+def update_pysilcam_offset(logger, settings, datafilename, datapath):
     '''
-    Administration of the -STATS.h5 file
+    Set the PYSILCAM_OFFSET based on stats file.
 
     Args:
         logger          (logger object) : logger object created using configure_logger()
         datafilename    (str)           : name of the folder containing the -STATS.h5
         datapath        (str)           : name of the path containing the data
 
     '''
-    if os.path.isfile(datafilename + '-STATS.h5'):
-        if overwriteSTATS:
-            logger.info('removing: ' + datafilename + '-STATS.h5')
-            print('Overwriting ' + datafilename + '-STATS.h5')
-            os.remove(datafilename + '-STATS.h5')
-        else:
-            logger.info('Loading old data from: ' + datafilename + '-STATS.h5')
-            print('Loading old data from: ' + datafilename + '-STATS.h5')
-            oldstats = pd.read_hdf(datafilename + '-STATS.h5', 'ParticleStats/stats')
-            logger.info('  OK.')
-            print('  OK.')
-            last_time = pd.to_datetime(oldstats['timestamp'].max())
-
-            logger.info('Calculating spooling offset')
-            print('Calculating spooling offset')
-
-            # TODO: move this import to the top
-            from pysilcam.fakepymba import silcam_name2time
-            files = [f for f in sorted(os.listdir(datapath))
-                     if f.endswith('.silc' or f.endswith('.bmp'))]
-            offsetcalc = pd.DataFrame(columns=['files', 'times'])
-            offsetcalc['files'] = files
-            for i, f in enumerate(files):
-                offsetcalc['times'].iloc[i] = silcam_name2time(f)
-            offset = int(min(np.argwhere(offsetcalc['times'] > last_time)))
-            # subtract the number of background images, so we get data from the correct start point
-            offset -= settings.Background.num_images
-            # and check offset is still positive
-            if offset < 0:
-                offset = 0
-            offset = str(offset)
-            os.environ['PYSILCAM_OFFSET'] = offset
-            logger.info('PYSILCAM_OFFSET set to: ' + offset)
-            print('PYSILCAM_OFFSET set to: ' + offset)
+
+    datafile_hdf = datafilename + '-STATS.h5'
+    add_metadata = False
+    logger.info('Loading old data from: ' + datafile_hdf)
+    print('Loading old data from: ' + datafile_hdf)
+    oldstats = pd.read_hdf(datafile_hdf, 'ParticleStats/stats')
+    logger.info('  OK.')
+    print('  OK.')
+    last_time = pd.to_datetime(oldstats['timestamp'].max())
+
+    logger.info('Calculating spooling offset')
+    print('Calculating spooling offset')
+
+    # TODO: move this import to the top
+    from pysilcam.fakepymba import silcam_name2time
+    files = [f for f in sorted(os.listdir(datapath))
+                if f.endswith('.silc' or f.endswith('.bmp'))]
```

Comment:
E127: continuation line over-indented for visual indent  <!-- Automatically posted by Lintly -->